### PR TITLE
feat: openToDate prop for Calendar

### PR DESCRIPTION
**Pull request by jacobdevera**
### Description
Adds `openToDate` prop to specify a date that will be focused when the Calendar is mounted if consumers desire to override the default, which is to focus today's date.

Official fiori spec: https://experience.sap.com/fiori-design-web/date-picker/#components

> You can change the default focus __ “today” to another date. This can save users several clicks when they create events. For an event end date, for example, the focus should propose a date in the future (after the start date).

### CODE REVIEW

File: src/DatePicker/DatePicker.js [hunk 1]
```diff
@@ -333,6 +333,7 @@ class DatePicker extends Component {
             localizedText,
             onBlur,
             onDatePickerClose,
+            openToDate,
             popoverProps,
```
Comment (prsdthkr) [inline src/DatePicker/DatePicker.js:337]: Should we add `openToDate` to `DatePicker.propTypes` too?

Comment (jacobdevera) [reply src/DatePicker/DatePicker.js:337]: I believe that is covered by `...Calendar.PropTypes`

Comment (prsdthkr) [reply src/DatePicker/DatePicker.js:337]: Right.

Wouldn't the de-structuring then be
```js
const { calendarProps: openToDate } = this.props
```
and story be
```js
<DatePicker calendarProps={{openToDate: _openToDate}} />;
```
?

Comment (jacobdevera) [reply src/DatePicker/DatePicker.js:337]: I think there is just a discrepancy for how we handle some Calendar props. Some of them are raw props to spread to the Calendar's HTML elements:
`calendarProps: PropTypes.shape({
        monthListProps: PropTypes.object,
        tableBodyProps: PropTypes.object,
        tableHeaderProps: PropTypes.object,
        tableProps: PropTypes.object,
        yearListProps: PropTypes.object
    })`

While the others that specify logic for dates and such (`blockedDates`, `disabledDates`, etc.) are pulled up a level to make for a simpler API and not having a huge `calendarProps` object.

Seems to be out of scope for this PR, though.

**Pull request merged**